cirq/experiments/fidelity_estimation_test.py

```diff
@@ -0,0 +1,89 @@
+import itertools
+
+from typing import Sequence
+
+import numpy as np
+import pytest
+
+import cirq
+
+MEASUREMENT_KEY = 'm'
+
+
+def sample_noisy_bitstrings(circuit: cirq.Circuit,
+                            qubit_order: Sequence[cirq.Qid],
+                            depolarization: float,
+                            n_samples: int) -> np.ndarray:
+    assert 0 <= depolarization <= 1
+    dim = np.product(circuit.qid_shape())
+    n_incoherent = int(depolarization * n_samples)
+    n_coherent = n_samples - n_incoherent
+    incoherent_samples = np.random.randint(dim, size=n_incoherent)
+    circuit_with_measurements = cirq.Circuit.from_ops(
+        circuit, cirq.measure(*qubit_order, key=MEASUREMENT_KEY))
+    if n_coherent > 0:
```

Comment:
I think this optimize belongs inside `sim.run` instead of outside of it. Also this is test code; don't worry about it.

Reply:
Filed #2154.